greetd: 0.9.0 -> 0.10.0

crftbt opened the pull request:
Updating greetd from 0.9.0 to 0.10.0 with sha256 from nurl. How do you get the cargoHash?

## Description of changes

https://git.sr.ht/~kennylevinsen/greetd/refs/0.10.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

crftbt committed:
greetd: 0.9.0 -> 0.10.0

Updating greetd from 0.9.0 to 0.10.0 with sha256 from nurl. How do you get the cargoHash?

RaghavSood commented:
Result of `nixpkgs-review pr 312468` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

RaghavSood commented on pkgs/applications/display-managers/greetd/default.nix:
```diff
@@ -8,13 +8,13 @@
+    sha256 = "sha256-QFxE5c4XDOBa3rKpUFur6xOG+5m0NSCOWo7960ZWEow=";
   };
 
   cargoHash = "sha256-vn1ih4YXsltR1zdy7BzS3CSJO9Uewx/654lRTSkWV0k=";
```
This probably needs to be updated

crftbt replied:
How do you get the cargoHash?

RaghavSood replied:
It does

```
error: hash mismatch in fixed-output derivation '/nix/store/5fci6drfrdpby4gd48chdbxp5159f66w-greetd-0.10.0-vendor.tar.gz.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-w7V5ZBFy5MIEQKOPPgsc7XnrXsF5rEDNzUKeLPPy4P4=
```

RaghavSood replied:
You give it a fake hash (just leave it blank), try to build it, then wait for the new one

crftbt replied:
I can set cargoHash to empty in a new github pull request, but I'm wondering where in the build process on github do you get that error message for the new hash?

RaghavSood replied:
You run `nix-build -A greetd` locally in your copy of the nixpkgs repo to test your changes and bulids.

RaghavSood replied:
It has to be inside your `nixpkgs` repo, you can't run it from an arbitrary point

I would suggest reading a guide on contributing to nixpkgs first - you can start with pkgs/README.md and CONTRIBUTING.md

crftbt committed:
greetd: 0.9.0 -> 0.10.0

Updating greetd from 0.9.0 -> 0.10.0.
Updating cargoHash, in addition to version and sha256.

crftbt reviewed (commented):
Cloned the repo locally and tested the build from there. Just edited the cargoHash. Is this sufficient, or should a new Pull Request be submitted? Your patience is appreciated.

RaghavSood commented:
You also need to squash your commits so it's only a single `greetd: 0.9.0 -> 0.10.0`, but looks correct now aside from that

crftbt commented:
I'm not seeing any options to squash the commits on GitHub. I will submit a new Pull Request with a single commit.

JohnAZoidberg commented:
Already updated in #318486
